crates/sui-core/src/authority.rs

```diff
@@ -1176,19 +1241,33 @@ impl AuthorityState {
             self.epoch() + 1 == new_committee.epoch,
             SuiError::from("Invalid new epoch to sign and update")
         );
-        self.database.sign_new_epoch(
+        let cur_epoch = new_committee.epoch;
+        let latest_epoch = self.epoch_store.get_latest_authenticated_epoch();
+        fp_ensure!(
+            latest_epoch.epoch() + 1 == cur_epoch,
+            SuiError::from("Unexpected new epoch number")
+        );
+
+        let signed_epoch = SignedEpoch::new(
             new_committee.clone(),
             self.name,
             &*self.secret,
             next_checkpoint,
-        )?;
+            latest_epoch.epoch_info(),
+        );
+        self.epoch_store
+            .epochs
+            .insert(&cur_epoch, &AuthenticatedEpoch::Signed(signed_epoch))?;
         // TODO: Do we want to make it possible to subscribe to committee changes?
```

Comment:
not related to this PR, but is there an issue for this TODO?

Reply:
Once I am done with all the MVP work, I will go through all TODOs and create issues for them